conans/client/tools/scm.py

```diff
@@ -146,4 +157,114 @@ def get_branch(self):
             branch = status.splitlines()[0].split("...")[0].strip("#").strip()
             return branch
         except Exception as e:
-            raise ConanException("Unable to get git branch from %s\n%s" % (self.folder, str(e)))
+            raise ConanException("Unable to get git branch from %s: %s" % (self.folder, str(e)))
+
+    def _check_git_repo(self):
+        try:
+            self.run("status")
+        except Exception:
+            raise ConanException("Not a valid git repository")
+
+
+class SVN(SCMBase):
+    cmd_command = "svn"
+    file_protocol = 'file:///' if platform.system() == "Windows" else 'file://'
+
+    def __init__(self, folder=None, runner=None, *args, **kwargs):
+        def runner_no_strip(command):
+            return decode_text(subprocess.check_output(command, shell=True))
+        runner = runner or runner_no_strip
+        super(SVN, self).__init__(folder=folder, runner=runner, *args, **kwargs)
+
+    def run(self, command):
+        # Ensure we always pass some params
+        return super(SVN, self).run(command="{} --no-auth-cache".format(command))
+
+    def checkout(self, url, revision="HEAD"):
+        output = ""
+        try:
+            self._check_svn_repo()
+        except ConanException:
+            params = " --trust-server-cert-failures=unknown-ca" if not self._verify_ssl else ""
+            output += self.run('co "{url}" . {params}'.format(url=url, params=params))
+        else:
+            assert url.lower() == self.get_remote_url().lower(), \
+                "%s != %s" % (url, self.get_remote_url())
+            output += self.run("revert . --recursive")
+        finally:
+            output += self.update(revision=revision)
+        return output
+
+    def update(self, revision='HEAD'):
+        self._check_svn_repo()
+        return self.run("update -r {rev}".format(rev=revision))
+
+    def excluded_files(self):
+        self._check_svn_repo()
+        excluded_list = []
+        output = self.run("status --no-ignore")
+        for it in output.splitlines():
+            if it[0] == 'I':  # Only ignored files
+                filepath = it[8:].strip()
+                excluded_list.append(os.path.normpath(filepath))
+        excluded_list.append(".svn")
```

Comment:
When git copies from local or remote location, the git folder keeps the `.git` folder in the source folder.
When svn uses a remote location, it keeps the `.svn` folder. But when copying it from local source, it removes the `.svn`folder. What the reason for excluding the `.svn´folder?
I need to generate a version header, which include some `svn info` parameters. And without being a valid subversion repository, there is no chance for me, to get these parameters.

Reply:
That behavior changed for Git in this commit (https://github.com/conan-io/conan/commit/85824d5fd7170b4c23d913e201335470e2c27a63#diff-beff9987dc9b09cf5c8bd12c509b9e02) but it was not noticed until now. Thanks!